chrome/browser/net/chrome_network_delegate.cc

```diff
@@ -327,6 +330,95 @@ int ChromeNetworkDelegate::OnBeforeURLRequest(
   return rv;
 }
 
+std::string generateBlockedElemhideJavaScript(std::string url) {
+  std::stringstream ss;
+
+  // basically copy-pasting JS from
+  // https://github.com/adblockplus/adblockpluschrome/blob/master/include.preload.js#L546
+  ss <<
+"function hideElement(element)\n\
+{\n\
+  function doHide()\n\
+  {\n\
+    let propertyName = \"display\";\n\
+    let propertyValue = \"none\";\n\
+    if (element.localName == \"frame\")\n\
+    {\n\
+      propertyName = \"visibility\";\n\
+      propertyValue = \"hidden\";\n\
+    }\n\
+\n\
+    if (element.style.getPropertyValue(propertyName) != propertyValue ||\n\
+        element.style.getPropertyPriority(propertyName) != \"important\")\n\
+      element.style.setProperty(propertyName, propertyValue, \"important\");\n\
+  }\n\
+\n\
+  doHide();\n\
+\n\
+  new MutationObserver(doHide).observe(\n\
+    element, {\n\
+      attributes: true,\n\
+      attributeFilter: [\"style\"]\n\
+    }\n\
+  );\n\
+}";
+
+  ss <<
+"\n\
+ var selector = \"[src='" + url + "'], [srcset='" + url + "']\";\n\
+ //console.log('blocked element selector: ' + selector);\n\
+ var elements = document.querySelectorAll(selector);\n\
+ if (elements.length > 0) {\n\
+   //console.log('found ' + elements.length + ' elemhide blocked elements for selector ' + selector);\n\
+   for (var e=0; e<elements.length; e++) {\n\
```

Comment:
What about using of `for..of` loops or even `forEach`?

Reply:
is it better in any way?
BTW in `adblock_bridge.cc` we iterate using index too.

Reply:
It's mainly consistent with other JS code (in core, e.g.). Additionally `document.querySelectorAll(selector).forEach(hideElement)` is much less code, what is also better.

Reply:
it's indeed more concise unless we want to output array size and have some additional checks (not just call some other function).

Reply:
i still don't see any point

Reply:
OK, however it conflicts with https://adblockplus.org/coding-style#javascript

Reply:
current code style assumes you have full debugging possibility. And if you "debug" using android logcat one have to have code that outputs debug values and does not require changing of iteration loop kind from "modern" if you need to see one value (array length).

Reply:
But one can still see the array length

Reply:
if we chain it like you suggest `document.querySelectorAll(selector).forEach(hideElement)` we can't do it

Reply:
I would suggest
```
let elements = document.querySelectorAll(selector);
if (elements.length > 0) {
  console.debug(`'found ${elements.length} elemhide blocked elements for selector '${selector}'`);
  for (let element of elements) {
    if (element.src == url)
      hideElement(element);
  }
}
```
BTW, we use `let` instead of `var`. Additionally, I would try to use `console.debug` and [template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) and rather somehow introduce the variable `url`.